src/assets/sass/protocol/components/_navigation.scss

```diff
@@ -58,79 +68,74 @@
     }
 
     @media #{$mq-md} {
-        margin: $spacing-lg 0;
+        @include bidi(((margin, $spacing-lg ($spacing-sm * 2) $spacing-lg 0, $spacing-lg 0 $spacing-lg ($spacing-sm * 2)),));
 
         a {
             margin-top: $spacing-sm;
         }
     }
 
     @media #{$mq-lg} {
+        @include bidi(((margin, $spacing-lg ($spacing-md * 2) $spacing-lg 0, $spacing-lg 0 $spacing-lg ($spacing-md * 2)),));
+
         a {
             @include background-size(112px, 32px);
             height: 32px;
             margin-top: $spacing-xs;
             width: 112px;
         }
     }
+
+    @media #{$mq-xl} {
+        @include bidi(((margin, $spacing-lg ($spacing-lg * 2) $spacing-lg 0, $spacing-lg 0 $spacing-lg ($spacing-lg * 2)),));
+    }
 }
 
 /* -------------------------------------------------------------------------- */
-// Menu & Download Button
+// Menu molecule
 
-.mzp-c-navigation-items {
-    clear: both;
-    padding-top: $spacing-sm;
+.mzp-c-navigation-menu {
+    @media #{$mq-md} {
+        @include bidi(((margin-left, -$spacing-sm, margin-right, 0),));
+        display: inline-block;
+        width: auto;
+    }
+
+    @media #{$mq-lg} {
+        @include bidi(((margin-left, -$spacing-md, margin-right, 0),));
+    }
+
+    @media #{$mq-xl} {
+        @include bidi(((margin-left, -$spacing-lg, margin-right, 0),));
+    }
+}
+
+/* -------------------------------------------------------------------------- */
+// Download button
+
+.mzp-c-navigation-download {
+    margin: $spacing-xl 0 $spacing-lg;
 
     .mzp-c-button-download {
         margin-bottom: 0;
     }
 
     @media #{$mq-md} {
         @include bidi((
-            (float, left, right),
-            (margin-left, $spacing-sm, margin-right, 0),
+            (float, right, left),
+            (margin, $spacing-lg 0 $spacing-lg $spacing-sm, $spacing-lg $spacing-sm $spacing-lg 0)
         ));
-        clear: none;
-        padding-top: 0;
-        width: calc(100% - (84px + #{$spacing-sm}));
-
-        .mzp-c-menu {
-            @include bidi(((float, left, right),));
-        }
-
-        .mzp-c-navigation-download {
-            display: none;
-        }
+        display: block;
+        margin: $spacing-lg 0;
+        max-width: 180px;
```

Comment:
The wrapping works really well without a max-width on the button.

Not a blocker though.